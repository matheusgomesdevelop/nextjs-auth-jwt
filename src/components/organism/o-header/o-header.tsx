import styles from "./o-header.module.scss";

// Atoms
import { A_Avatar } from "@/components/atoms";

export const O_Header = () => {
  return (
    <header data-testid="o-header" className={styles.o_header}>
      <A_Avatar image={"/images/a-avatar.jpeg"} label={""} shape="circle" />
    </header>
  );
};
