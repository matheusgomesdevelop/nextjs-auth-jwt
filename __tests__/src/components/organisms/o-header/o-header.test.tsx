import renderer from "react-test-renderer";

// Organism
import { O_Header } from "@/components/organism/o-header/o-header";

describe("Deve renderizar o Header corretamente", () => {
  it("Deve preservar a estrutura visual do componente", () => {
    const three = renderer.create(<O_Header />).toJSON();

    expect(three).toMatchSnapshot();
  });
});
