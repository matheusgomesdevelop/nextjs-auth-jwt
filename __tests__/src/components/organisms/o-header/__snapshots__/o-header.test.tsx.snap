// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Deve renderizar o Header corretamente Deve preservar a estrutura visual do componente 1`] = `
<header
  className="o_header"
  data-testid="o-header"
>
  <div
    className="p-avatar p-component p-avatar-image p-avatar-circle"
    data-pc-name="avatar"
    data-pc-section="root"
    data-testid="a-avatar"
    style={{}}
  >
    <img
      alt="avatar"
      data-pc-section="image"
      onError={[Function]}
      src="/images/a-avatar.jpeg"
    />
  </div>
</header>
`;
